### PR TITLE
[GH-61] Use SiteURL for all calls from the webapp

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
**Summary** 

- Used site URL for all the API calls from the webapp to the server.
- This PR is an extension from PR:- https://github.com/mattermost/mattermost-plugin-skype4business/pull/78

- [Issue #61](https://github.com/mattermost/mattermost-plugin-skype4business/issues/61)

### CODE REVIEW

File: webapp/src/client/client.js [hunk 1]
```diff
@@ -124,14 +124,15 @@ AuthenticationContext.prototype._loginPopup = function _loginPopup(urlNavigate,
 };
 
 export default class Client {
-    constructor() {
+    setServerRoute(url) {
+        this.url = url + `/plugins/${pluginID}/api/v1`;
```
Comment (javaguirre) [inline webapp/src/client/client.js:128]:
```suggestion
        this.url = `${url}/plugins/${pluginID}/api/v1`;
```

File: webapp/src/selectors/index.js [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
 import {id as pluginId} from '../manifest';
 
-const getPluginState = (state) => state['plugins-' + pluginId] || {};
+const getPluginState = (state) => state[`plugins-${pluginId}`] || {};
+
+export const getPluginServerRoute = (state) => {
+    const config = getConfig(state);
+
+    let basePath = '';
+    if (config && config.SiteURL) {
```
Comment (javaguirre) [inline webapp/src/selectors/index.js:11]: can't you do `config?.SiteURL` ?

File: webapp/src/selectors/index.js [hunk 3]
```diff
@@ -1,10 +1,22 @@
...
+        basePath = new URL(config.SiteURL).pathname;
 
-const getAuthenticationResult = (state) => getPluginState(state).authenticationResult;
+        if (basePath && basePath[basePath.length - 1] === '/') {
+            basePath = basePath.substr(0, basePath.length - 1);
```
Comment (javaguirre) [inline webapp/src/selectors/index.js:15]: `basePath.replace(/\/$/, "");` is more straight forward IMO.

**Review by javaguirre — CHANGES_REQUESTED** [review]
Thank you! some small details.

**Review by javaguirre — APPROVED** [review]
Great work

**Comment by hanzei** [comment]
@Kshitij-Katiyar Would you please merge `master` into your branch?

**Comment by Kshitij-Katiyar** [comment]
> @Kshitij-Katiyar Would you please merge `master` into your branch?

@hanzei synced with the master

**Review by mickmister — APPROVED** [review]

**Comment by hanzei** [comment]
@DHaussermann Would you mind testing the PR in release testing?

**Pull request merged**